tasks.py

```diff
@@ -856,3 +925,300 @@ def tests(context, lint_only=False, keepdb=False):
+                    tests_times.remove(item)
+
+        if tests_times:
+            raise ValueError(f"Tests not found in _NAMES_DISTRIBUTION: {tests_times}")
+
+        avg = sum(estimations) / tests_times_len
+
+        for index, name in enumerate(names):
+            for filename in all_tests_files[:]:
+                if filename.startswith(f"nautobot/{name}"):
+                    print(f"WARNING: Test not found in tests_times, adding: {filename}")
+                    obsoleted.append(filename)
+                    add_file(filename, avg, index)
+    elif distribution == DistributionType.TIMES:
+        # Sort tests by time from slowest to fastest
+        tests_times.sort(key=lambda item: item["time"], reverse=True)
+        for file in tests_times:
+            # Add the next slowest test to the group with the lowest total times
+            add_file(file["file"], file["time"])
+
+        avg = sum(estimations) / len(tests_times)
+
+        for filename in all_tests_files[:]:
+            print(f"WARNING: Test not found in tests_times, adding: {filename}")
+            obsoleted.append(filename)
+            add_file(filename, avg)
+
+        # Run faster tests first to fail fast
+        for modules in groups:
+            modules.reverse()
+    else:
+        raise NotImplementedError("Only NAMES or TIMES distribution is supported at this time.")
+
+    if all_tests_files:
+        for filename in all_tests_files:
+            print(f"ERROR: Not distributed: {filename}")
+        raise RuntimeError("Not all tests were distributed, please check the logs for more information.")
+
+    sorted_groups = sorted(groups, key=lambda group: estimations[groups.index(group)], reverse=True)
+    sorted_estimations = [estimations[groups.index(group)] for group in sorted_groups]
+
+    if obsoleted:
+        print(f"WARNING: {_TESTS_TIMES_PATH} is obsolete, please update it by running `inv sum-tests-times`.")
+
+    return sorted_groups, sorted_estimations
+
+
+@task(
+    name="unittest-parallel",
+    help={
+        "workers": "Number of parallel workers to use (default: 3)",
+        "default-exec": "Default docker compose execution method 'local', 'exec' or 'run' (default: 'exec')",
+        "distribution": "Distribution method `names` or `times` (default: `names`)",
+    },
+)
+def unittest_parallel(context, workers=3, default_exec=None, distribution="names"):
+    """Parallelize unit tests."""
+    start_time = time.time()
+
+    if default_exec is None:
+        default_exec = "local" if is_truthy(context.nautobot.local) else "exec"
+    if default_exec != "local":
+        stop(context)
+        if default_exec == "exec":
+            docker_compose(context, "up --detach -- nautobot")
+        elif default_exec == "run":
+            docker_compose(context, "up --detach -- db redis selenium")
+        else:
+            raise ValueError(f"Invalid --default-exec: {default_exec}")
```

Comment:
This is being done to ensure that the worker container is *not* running? Or some other reason? A comment would be appreciated. :)

Reply:
Comment with explanation added.